### ISSUE TITLE
Failed to save 3D textures of RGB32F/UI/I as dds files

### ISSUE THREAD

**Issue by dreamelon** [baldurk/renderdoc#3037]
## Description

I'm debugging GL_Texture_Zoo on AMD cards. I tried to save 3D texture of RGB32F as a DDS file and then opened it, it looked normally with slice 0 but corrupted with slice 0+. I encountered an assertion when opening the file in line 160 in streamio.h

Correct
![image](https://github.com/baldurk/renderdoc/assets/26594643/a4dbd238-16f1-446d-a0f6-ec01a401ad23)

Corrupted
![image](https://github.com/baldurk/renderdoc/assets/26594643/eb03c953-819a-487c-9480-bc724c7734dd)

## Steps to reproduce

1. Get a trace about GL_Texture_Zoo
2. Locate to the event RGB32F and then sub-event Texture 3D, save the binding texture as a dds file.
3. Open the dds again and switch slice, you'll see the corruption 

## Environment

* RenderDoc version: v.128
* Operating System: Windows 11 22H2
* Graphics API: OpenGL
* GPU: AMD 6700XT

**Comment by baldurk** [comment]
I can't reproduce this given these steps. I tried this on several systems on both NV and AMD on latest drivers as well as on linux mesa, and the 3D RGB32F texture saves and loads correctly for me.

Can you try to reproduce this on a different system? In particular can you reproduce this on a non-AMD system as there are a number of driver bugs in this area on AMD so it wouldn't surprise me if this is a driver issue and not anything on RenderDoc's side.

If you can reproduce this on a multiple systems can you share the dds file? that would tell us if the problem is either in the saving of the dds or the loading of it. Similarly, try saving each individual slice to a png - that would tell if the problem is in fetching the data from the GPU or encoding to the target file format.

**Comment by dreamelon** [comment]
[capture_frame5.zip](https://github.com/baldurk/renderdoc/files/12428260/capture_frame5.zip)
Hi, 
Could you please try this trace? It can correctly save each slice to a PNG. I suspect the root cause is in the loading of it. Does renderdoc use D3D11 default to load dds file? Because I don't see obvious error on OGL driver side and also there is a similar failure when using D3D proxy.

**Comment by baldurk** [comment]
I also can't reproduce it with that capture, it works fine for me on NV and AMD. That capture was not produced from the normal GL_Texture_Zoo as it looks like you have made modifications to the program. Similarly it wasn't made by the stable version of v1.28 - can you please test on the stable v1.28 so we know we are running the same setup as much as can be the same.

Please be careful when giving your reproduction steps. If you say to run one program with one version but you are running a different program with a different version, that can easily cause confusion and wasted effort on both sides if your changes are significant.

You haven't answered all of my questions from the previous comment. Can you reproduce this on other systems in particularly on a non-AMD system? Can you share the dds file itself? E.g. here's a zip containing the dds I saved from your capture and from a normal capture of GL_Texture_Zoo and they open correctly for me: [dds_files.zip](https://github.com/baldurk/renderdoc/files/12428444/dds_files.zip)

RenderDoc always loads images using D3D11 on windows, GL is only used on linux where D3D11 isn't available.

**Comment by Zorro666** [comment]
Using your capture and/or a local capture of GL_Texture_Zoo have reproduced this bug when replaying on relatively modern AMD GPU (discrete and integrated and up to date AMD drivers).

Loading the same DDS file displays correctly on nVidia GPU or using the WARP driver and also with a local workaround to avoid using UpdateSubresource().

This looks to be an AMD driver issue (and not a RenderDoc code issue)

**Comment by dreamelon** [comment]
Sorry for not making the issue clear.
I did see the issue in stable v1.28 too.  I wanted to simplify this issue so made a lot of modifications to the test.

I just got a NV machine to test and did not see any corruption.

>This looks to be an AMD driver issue (and not a RenderDoc code issue)
>RenderDoc always loads images using D3D11 on windows, GL is only used on linux where D3D11 isn't available.

I agree and actually I'm investigating it. I dont know which api renderdoc uses so not sure which driver is broken. Your information is helpful.
Thanks for your quick response!
